### PR TITLE
Fix #139: Add code reset button to page

### LINKED ISSUE #139: No reset button to reset the current code
@sniffsky and I are currently working on #98 , however, we realize that if we are able to save codes, then it's probably good to have a reset button once #98 is resolved.

### PR DESCRIPTION
**Pull request by wmcalyj**
This change contains both UI change (new button added) and backend change (code are removed from localStorage)

### CODE REVIEW

**Comment by seanlip** [comment]
Code looks good, thanks! Deferring to @rabidbit for UI check.

File: client/components/LearnerViewDirective.js [hunk 1]
```diff
@@ -142,7 +146,7 @@ tie.directive('learnerView', [function() {
           -webkit-font-smoothing: antialiased;
         }
         .tie-coding-terminal:focus, .tie-lang-select-menu:focus,
-            .tie-run-button:focus {
+            tie-code-reset:focus, .tie-run-button:focus {
```
Comment (rabidbit) [inline client/components/LearnerViewDirective.js]: **.**tie-code-reset:focus ?

Comment (wmcalyj) [inline client/components/LearnerViewDirective.js]: Removed that part.

File: client/components/LearnerViewDirective.js [hunk 2]
```diff
@@ -67,6 +67,10 @@ tie.directive('learnerView', [function() {
                 <select class="tie-lang-select-menu" name="lang-select-menu">
                   <option value="Python" selected>Python</option>
                 </select>
+                <button class="tie-code-reset" name="code-reset"
+                    ng-click="resetCode()">
+                  Reset
```
Comment (rabidbit) [inline client/components/LearnerViewDirective.js]: Change "Reset" to "Reset Code". This is to avoid confusion about what exactly the button is resetting.

Comment (wmcalyj) [inline client/components/LearnerViewDirective.js]: Done

File: client/components/LearnerViewDirective.js [hunk 3]
```diff
@@ -214,6 +218,11 @@ tie.directive('learnerView', [function() {
           float: left;
           margin-top: 10px;
         }
+        .tie-code-reset {
```
Comment (rabidbit) [inline client/components/LearnerViewDirective.js]: If it doesn't break the CSS, please insert alphabetically (after .tie-arrow-highlighter)

**Comment by rabidbit** [comment]
Added comments. Please send in an updated commit with the fixes. Thanks!

File: client/components/LearnerViewDirective.js (hunk 3 above)
Comment (wmcalyj) [inline client/components/LearnerViewDirective.js]: Done

**Comment by wmcalyj** [comment]
@rabidbit Addressed review comments with the previous commits. PTAL

**Comment by rabidbit** [comment]
See it. Thanks!

**Comment by wmcalyj** [comment]
Thank you John!

**Pull request merged**